src/Jackett.Common/Utils/StringUtil.cs

```diff
@@ -138,13 +138,35 @@ public static string MakeValidFileName(string text, char? replacement = '_', boo
             return changed ? sb.ToString() : text;
         }
 
-        public static string GetQueryString(this NameValueCollection collection, Encoding encoding = null) =>
-            string.Join("&", collection.AllKeys.Select(a =>
-                $"{a}={WebUtilityHelpers.UrlEncode(collection[a], encoding ?? Encoding.UTF8)}"));
+        /// <summary>
+        /// Converts a NameValueCollection to an appropriately formatted query string.
+        /// Duplicate keys are allowed in a NameValueCollection, but are stored as a csv string in Value.
+        /// This function handles leaving the values together in the csv string or splitting the value into separate keys
+        /// </summary>
+        /// <param name="collection">The NameValueCollection being converted</param>
+        /// <param name="encoding">The Encoding to use in url encoding Value</param>
+        /// <param name="splitMultiValues">Whether the Value should be Key=Value or split into Key=Val1&Key=Val2</param>
+        /// <param name="separator">The string used to separate each query value</param>
+        /// <returns>A web encoded string of key=value parameters separated by the separator</returns>
+        public static string GetQueryString(this NameValueCollection collection, Encoding encoding = null,
```

Comment:
It's possible to extend NameValueCollection instead?

Reply:
Not sure what you're asking? NameValueCollection is a Microsoft built-in class in System.Collections.Specialized.

Reply:
I explained bad myself. How this will be called?
`var qc = new NameValueCollection ...    qc.GetQueryString()`
or
`var qc = new NameValueCollection ...    StringUtil.GetQueryString(qc)`
I don't like the 2nd one, it's error prone.

Reply:
This is the extension method we've been using in this pattern:
```
var queryParams = new NameValueCollection
{ ... };
var searchLink = SearchLink + "?" + queryParams.GetQueryString();
```

So to answer your question, it is called in the first manner. Although technically you can call it the second way too, because the compiler is converting the first to the second behind the scenes. Any static function in a static class whose first parameter is `this var name` can be used like this.

Reply:
ohh, I like this :)